### DISCUSSION TITLE
MatMul at the input not getting Converted to MatrixVectorActivation layer after convert_to_hls step in streamlining

### DISCUSSION THREAD

**Discussion by shashwat1198** [Xilinx/finn#1073] [Q&A]
Hello,

I had a question regarding why a MatMul node at the input of the graph is not getting Converted to MatrixVectorActivation layer after convert_to_hls step during hardware generation. 
Due to this, this Matmul does not get included in the dataflow partition created by FINN and hence not included in the final hardware design.

![step_convert_to_hls onnx_no_input_quant](https://github.com/Xilinx/finn/assets/49102578/40d253d9-5d2c-4325-92d4-56bab582c232)
![dataflow_parent_no_input_quant onnx](https://github.com/Xilinx/finn/assets/49102578/df17ddec-6884-4feb-8a06-4ad4c2a03240)

I have attached the _convert_to_hls_ model_ onnx file and the onnx model of _dataflow_parent_ file step here. Any pointers to get the Matmul layer inside the partition would be very helpful. 

P.S : When I set the input quantizer to INT8 then that MatMul layer gets absorbed in the dataflow partition that FINN creates but one Multithreshold node is still excluded from the partition in that implementation as well. I am attaching the onnx files for this variant here as well.

![step_convert_to_hls_with_input_quant onnx](https://github.com/Xilinx/finn/assets/49102578/86556db0-24c2-490d-9de6-0e5d0507f9ab)
![dataflow_parent_with_input_quant onnx](https://github.com/Xilinx/finn/assets/49102578/6bd6ced8-dc8c-43b1-b5bb-726e4b72b7e6)

So basically what should I do to make sure all nodes that get the input are inferred as some hardware block in the _convert_to_hls_ step?
